linked_lists/intersection.py

```diff
@@ -11,4 +11,11 @@ def intersection_node(headA, headB):
     """ Will return the node at which the two lists intersect.
         If the two linked lists have no intersection at all, return None.
     """
-    pass
\ No newline at end of file
+    if not headA or not headB:
+        return None
+    p1, p2 = headA, headB
+    while p1.val!= p2.val:
```

Comment:
We want to compare the node _objects_ and not just their _values_. Two nodes can have the same values but be different objects in memory!
```suggestion
    while p1 != p2:
```